sktime/transformations/series/dobin.py

```diff
@@ -0,0 +1,215 @@
+            warnings.warn(
+                "Warning: Input data X is univariate. For dimensionality reduction, "
+                "please provide multivariate input."
+            )
+            self._coords = X
+            return self
+
+        # if more dimensions than observations, change of basis to subspace
+        if n_obs < n_dim:
+            pca = PCA(n_components=n_obs)
+            X = pca.fit_transform(X)
+            self._X_pca = X
+            _, n_dim = X.shape
+
+        self.k_ = min(20, max(n_obs // 20, 2)) if self.k is None else self.k
+
+        X_copy = X.copy()
+        B = np.identity(n_dim)
+        basis = pd.DataFrame()
+
+        for _ in range(n_dim):
+            # Compute Y space
+            y_space = close_distance_matrix(X_copy, self.k_, self.frac)
+
+            # Find eta
+            w = y_space.apply(sum, axis=0)
+            eta = np.array([w / np.sqrt(sum(w**2))])
+
+            # If issues finding Y space (e.g. no variance in column)
+            # get null space of basis
+            if np.isnan(eta).any():
+                basis_col = pd.DataFrame(null_space(basis.T))
+                basis = pd.concat([basis, basis_col], axis=1)
+                break
+
+            # Update basis
+            basis_col = pd.DataFrame(np.dot(B, eta.T))
+            basis = pd.concat([basis, basis_col], axis=1)
+
+            # Find xperp
+            xperp = X_copy - np.dot(np.dot(np.array(X_copy), eta.T), eta)
+
+            # Find a basis B for xperp
+            B1 = null_space(eta)
+
+            # Change xperp coordinates to B basis
+            X_copy = np.dot(xperp, B1)
+
+            # Update B with B1, each time 1 dimension is reduced
+            B = np.dot(B, B1)
+
+        # new coordinates
+        coords = pd.DataFrame(
+            X.dot(np.array(basis))
+        )  # convert np.array, error if both pd.DataFrames and rownames != colnames
+
+        basis.columns = ["".join(["DB", str(i)]) for i in range(len(basis.columns))]
+        self._basis = basis
+        coords.columns = ["".join(["DB", str(i)]) for i in range(len(coords.columns))]
+        self._coords = coords
+
+        return self
+
+    def _transform(self, X, y=None):
+
+        # fit again if indices not seen, but don't store anything
+        if not X.index.equals(self._X.index):
+            X_full = X.combine_first(self._X)
+            new_dobin = DOBIN(
+                frac=self.frac,
+                normalize=self.normalize,
+                k=self.k,
+            ).fit(X_full)
+            warnings.warn(
+                "Warning: Input data X differs from that given to fit(). "
+                "Refitting with new input data, not storing updated public class "
+                "attributes. For this, explicitly use fit(X) or fit_transform(X)."
+            )
+            return new_dobin._coords
+
+        return self._coords
+
+
+def close_distance_matrix(X, k, frac):
+    """Calculate distance between close pairs."""
```

Comment:
Could you expand the docstring to cover argument types and descriptions and the output?